### DISCUSSION TITLE
How to get RibbonButton properties?

### DISCUSSION THREAD

**Discussion by rulai-yinqiang** [harborsiem/WindowsRibbon#23] [Q&A]
Hello,

I can use this code to get a button, and can AddHandler and reset Label or Image in Form_Load event.

`_buttonDropB = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButtonDropB))`

But all the control propertiese are Nothing. How do I get control Label, Name etc. properties.

> {RibbonLib.Controls.RibbonButton}
>     CommandID: 1009
>     CommandType: Unknown {0}
>     Enabled: False
>     Keytip: Nothing
>     Label: Nothing
>     LabelDescription: Nothing
>     LargeHighContrastImage: Nothing
>     LargeImage: Nothing
>     Name: ""
>     SmallHighContrastImage: Nothing
>     SmallImage: Nothing
>     Tag: Nothing
>     TooltipDescription: Nothing
>     TooltipTitle: Nothing
>     _commandID: 1009
>     _mapEvents: Count = 1
>     _mapProperties: Count = 10
>     _ribbon: {RibbonLib.Ribbon}
> 
>

**Comment by harborsiem** [comment]
Please have a look to the Wiki (News and Hints) for:
Keytip: Nothing
Label: Nothing
LabelDescription: Nothing
LargeHighContrastImage: Nothing
LargeImage: Nothing
SmallHighContrastImage: Nothing
SmallImage: Nothing
TooltipDescription: Nothing
TooltipTitle: Nothing

Following properties you have to set by code first:
Name: ""
Tag: Nothing

Following properties are updated by the Framework when the RibbonButton is first time in view.
CommandType: Unknown {0}
Enabled: False
